x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticSparseVectorQueryRewriteInterceptor.java

```diff
@@ -0,0 +1,99 @@
+        QueryBuilder rewritten = queryBuilder;
+        InferenceIndexInformationForField inferenceIndexInformationForField = SemanticQueryInterceptionUtils.resolveIndicesForField(
+            sparseVectorQueryBuilder.getFieldName(),
+            context.getResolvedIndices()
+        );
+
+        if (inferenceIndexInformationForField == null || inferenceIndexInformationForField.inferenceIndices().isEmpty()) {
+            // No inference fields, return original query
+            return rewritten;
+        } else if (inferenceIndexInformationForField.nonInferenceIndices().isEmpty() == false) {
+            // Combined inference and non inference fields
+            BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+            boolQueryBuilder.should(
+                SemanticQueryInterceptionUtils.createSubQueryForIndices(
+                    inferenceIndexInformationForField.nonInferenceIndices(),
+                    SemanticQueryInterceptionUtils.createSubQueryForIndices(
+                        inferenceIndexInformationForField.nonInferenceIndices(),
+                        sparseVectorQueryBuilder
+                    )
+                )
+            );
+            // We always perform nested subqueries on semantic_text fields, to support
+            // sparse_vector queries using query vectors
+            boolQueryBuilder.should(
+                SemanticQueryInterceptionUtils.createSubQueryForIndices(
+                    inferenceIndexInformationForField.inferenceIndices(),
+                    buildNestedQueryFromSparseVectorQuery(sparseVectorQueryBuilder)
+                )
+            );
+            rewritten = boolQueryBuilder;
+        } else {
+            // Only semantic text fields
+            rewritten = buildNestedQueryFromSparseVectorQuery(sparseVectorQueryBuilder);
+        }
+
+        return rewritten;
+    }
+
+    private QueryBuilder buildNestedQueryFromSparseVectorQuery(SparseVectorQueryBuilder sparseVectorQueryBuilder) {
+        return QueryBuilders.nestedQuery(
+            getNestedFieldPath(sparseVectorQueryBuilder.getFieldName()),
+            new SparseVectorQueryBuilder(
+                getNestedEmbeddingsField(sparseVectorQueryBuilder.getFieldName()),
+                sparseVectorQueryBuilder.getQueryVectors(),
+                sparseVectorQueryBuilder.getInferenceId(),
+                sparseVectorQueryBuilder.getQuery(),
+                sparseVectorQueryBuilder.shouldPruneTokens(),
+                sparseVectorQueryBuilder.getTokenPruningConfig()
+            ),
+            ScoreMode.Max
+        );
+    }
+
+    private static String getNestedFieldPath(String fieldName) {
+        return String.join(".", fieldName, SemanticTextField.INFERENCE_FIELD, SemanticTextField.CHUNKS_FIELD);
+    }
+
+    private static String getNestedEmbeddingsField(String fieldName) {
+        return String.join(".", getNestedFieldPath(fieldName), SemanticTextField.CHUNKED_EMBEDDINGS_FIELD);
+    }
```

Comment:
No need for these methods, you can use `SemanticTextField#getChunksFieldName` and `SemanticTextField#getEmbeddingsFieldName`